detach/detach_example_test.go

```diff
@@ -0,0 +1,172 @@
+package detach_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/reddit/baseplate.go/detach"
+)
+
+func init() {
+	detach.Register(detach.Hooks{
+		Inline: func(dst, src context.Context) context.Context {
+			if v, ok := src.Value(contextKey).(*ctxVal); ok && v != nil {
+				dst = context.WithValue(dst, contextKey, v)
+			}
+			return dst
+		},
+		Async: func(dst, src context.Context, next func(ctx context.Context)) {
+			if v, ok := src.Value(contextKey).(*ctxVal); ok && v != nil {
+				newVal := &ctxVal{val: v.val}
+				dst = context.WithValue(dst, contextKey, newVal)
+				defer func() {
+					fmt.Printf("newVal.closed() == %v\n", newVal.closed)
+					newVal.close()
+					fmt.Printf("newVal.closed() == %v\n", newVal.closed)
```

Comment:
did you mean for one of these prints to be outside of the defer?

Reply:
No, the point was to show that it wasn't closed even if the one on the original context was and then to show that it was closed.